### ISSUE TITLE
qitmeer crash by nil pointer dereference

### ISSUE THREAD

**Issue by dindinw** [Qitmeer/qitmeer#477]
### Qitmeer version
qitmeer version 0.10.0+dev-cc570c5 (Go version go1.16)

although the issued version `cc570c` is not the latest version `5ddabc`, but the nil pointer issue should exist before `f2305f`, see #474 also.
 
- https://github.com/Qitmeer/qitmeer/commit/5ddabc2b32301d66c570add5a6b8b6e09542cb68
- https://github.com/Qitmeer/qitmeer/commit/cc570c526062facf2cdc7c90207ce89c24539fea
- https://github.com/Qitmeer/qitmeer/commit/f2305f24549b70b643e11330493d725ea96ba52c

### Error log
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1b4121c]

goroutine 3314967 [running]:
github.com/Qitmeer/qitmeer/p2p/peers.(*Peer).GraphState(0x0, 0x0)
        /Users/alex/work/qitmeer/qitmeer/p2p/peers/peer.go:373 +0x3c
github.com/Qitmeer/qitmeer/p2p/synch.(*PeerSync).IsCompleteForSyncPeer(0xc00025d000, 0xc0009ae001)
        /Users/alex/work/qitmeer/qitmeer/p2p/synch/peersync.go:356 +0x57
github.com/Qitmeer/qitmeer/p2p/synch.(*PeerSync).IsCurrent(0xc00025d000, 0x8737220222786)
        /Users/alex/work/qitmeer/qitmeer/p2p/synch/peersync.go:344 +0x5a
github.com/Qitmeer/qitmeer/p2p/synch.(*Sync).handleInventory(0xc000288870, 0xc000648dc0, 0xc000f00000, 0x21337b8, 0xc000648e40)
        /Users/alex/work/qitmeer/qitmeer/p2p/synch/inventory.go:73 +0x5b
github.com/Qitmeer/qitmeer/p2p/synch.(*Sync).inventoryHandler(0xc000288870, 0x2133828, 0xc0028fa420, 0x1e260e0, 0xc000648dc0, 0x21406e8, 0xc00f7875c0, 0x0)
        /Users/alex/work/qitmeer/qitmeer/p2p/synch/inventory.go:58 +0x11d
github.com/Qitmeer/qitmeer/p2p/synch.RegisterRPC.func1(0x21406e8, 0xc00f7875c0)
        /Users/alex/work/qitmeer/qitmeer/p2p/synch/sync.go:255 +0x74a
github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).SetStreamHandler.func1(0xc004fcf890, 0x23, 0x47b7360, 0xc00f7875c0, 0x1, 0x0)
        /Users/alex/go/pkg/mod/github.com/libp2p/go-libp2p@v0.11.0/p2p/host/basic/basic_host.go:566 +0xa2
created by github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).newStreamHandler
        /Users/alex/go/pkg/mod/github.com/libp2p/go-libp2p@v0.11.0/p2p/host/basic/basic_host.go:416 +0x665
```

**Comment by dindinw** [comment]
might related with #474

**Comment by dindinw** [comment]
After upgraded to the latest version `5ddabc2` and restarted the crashed qitmeer, the consensus lost.

### The consensus lost

*2021-04-16|06:58:40.633 [INFO |/services/common/progresslog/blocklogger.go:111] Processed 1 block in the last 12.45s (1 transaction, **order uncertainty**, 2021-04-16 02:23:40 +0800 CST)module=blkmanager*

The block DAG into a error status at `[594,96351,96413]`
```
594,96351,96413,16Uiu2HAmVMUdpBogwrZKc36RwMkiHdNBYvczNif6sT8398rk3kfY,100000,0.10.0+dev-5ddabc2,30,local
```

### Log
```
2021-04-16|06:58:28.000 [INFO |qitmeerd/qitmeerd.go:62]      System info                        Qitmeer Version=0.10.0+dev-5ddabc2 Go version=go1.16
2021-04-16|06:58:28.000 [INFO |qitmeerd/qitmeerd.go:63]      System info                        Home dir=/Users/alex/data/qitmeer/0.10-local
2021-04-16|06:58:28.000 [INFO |/services/common/db.go:29]    Loading block database             dbPath=/Users/alex/data/qitmeer/0.10-local/data/mixnet/blocks_ffldb
2021-04-16|06:58:28.035 [INFO |/database/ffldb/reconcile.go:91] Detected unclean shutdown - Repairing...module=database
2021-04-16|06:58:28.039 [INFO |/database/ffldb/reconcile.go:96] Database sync complete             module=database
2021-04-16|06:58:28.039 [INFO |/services/common/db.go:49]       Block database loaded
2021-04-16|06:58:28.127 [INFO |/p2p/options.go:110]             Start Peers from:/Users/alex/data/qitmeer/0.10-local/data/mixnet/peerstoremodule=P2P
2021-04-16|06:58:28.170 [INFO |/node/node.go:131]               Starting Server                    module=node
2021-04-16|06:58:28.172 [INFO |/node/full.go:112]               Transaction index is enabled       module=node
2021-04-16|06:58:28.177 [INFO |/node/full.go:116]               Address index is enabled           module=node
2021-04-16|06:58:28.177 [INFO |/core/blockdag/phantom.go:43]    anticone size:6                    module=blockdag
2021-04-16|06:58:28.189 [INFO |/core/blockchain/blockchain.go:496] Loading dag ...                    module=blockchain
2021-04-16|06:58:29.772 [INFO |/core/blockchain/blockchain.go:510] Dag loaded:loadTime=1.582815041s   module=blockchain
2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:362] DAG Type:phantom                   module=blockchain
2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:363] Blockchain database version        module=blockchain chain=7 compression=1 index=1
2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:367] Chain state:totaltx=98436 tipsNum=5 mainOrder=98410 total=98418module=blockchain
2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:371] hash=665b9e86511c01fc1ecf3fba77c384c0a3d4c355b316f2c0e43de9104d73c5bd,order=uncertainty,work=13021961124926484249272015module=blockchain
2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:371] hash=c21f2f114e84ed5ecc05325770abcbe9d0f517342e171e2e4d0f939524e8c5a3,order=uncertainty,work=13040954545946961780708651module=blockchain
2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:371] hash=00000265f4cffd307482749124ac7f905fe5a0b785d8437032979284fc208504,order=98410,work=13046688408896539903688257module=blockchain
2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:371] hash=00888629ae9379c263eeb1fff343484cdc3efdf2a3d6acc87bc4dd6fe62b63d3,order=uncertainty,work=12813391860135580225320356module=blockchain
2021-04-16|06:58:32.536 [INFO |/core/blockchain/blockchain.go:371] hash=a1a18b16cbc72e980ebd9e9910178f483ac826757134c6bf9115deae5d65edce,order=uncertainty,work=12926277286955399406386273module=blockchain
2021-04-16|06:58:32.536 [INFO |/services/zmq/zmqdefaultnotify.go:17] ZMQ:Not Supported                  module=zmq
2021-04-16|06:58:32.536 [INFO |/services/tx/txmanager.go:40]         Starting tx manager                module=txmanager
2021-04-16|06:58:32.536 [INFO |/p2p/service.go:93]                   P2P Service Start                  module=P2P
2021-04-16|06:58:32.536 [INFO |/p2p/synch/peersync.go:50]            P2P PeerSync Start                 module=SYNC
2021-04-16|06:58:32.580 [INFO |/p2p/service.go:701]                  Node started p2p server:multiAddr=/ip4/10.0.0.6/tcp/28130/p2p/16Uiu2HAmVMUdpBogwrZKc36RwMkiHdNBYvczNif6sT8398rk3kfYmodule=P2P
2021-04-16|06:58:32.580 [INFO |/p2p/rebroadcast.go:38]               Starting Rebroadcast               module=P2P
2021-04-16|06:58:32.593 [INFO |/rpc/rpcserver.go:247]                RPC server listening on addr:0.0.0.0:10101module=RPC
2021-04-16|06:58:35.831 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAmKxB76C3dhkyRw18pZiRQo8Tm5mVwrVidLQtvsNHoUUN8 direction:Outbound multiAddr:/ip4/185.186.147.152/tcp/28130/p2p/16Uiu2HAmKxB76C3dhkyRw18pZiRQo8Tm5mVwrVidLQtvsNHoUUN8 activePeers:1 Peer Connectedmodule=SYNC
2021-04-16|06:58:35.831 [INFO |/p2p/synch/peersync.go:262]           Syncing to state (99006,96351,96413,99022,2) from peer 16Uiu2HAmKxB76C3dhkyRw18pZiRQo8Tm5mVwrVidLQtvsNHoUUN8 cur graph state:(98410,95813,95874,98418,1)module=SYNC
2021-04-16|06:58:36.374 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAkvMptnSRza7PoZGGTPjZfbXC6KzYjEcy21R8JEgUnkd1C direction:Outbound multiAddr:/ip4/149.248.8.175/tcp/28130/p2p/16Uiu2HAkvMptnSRza7PoZGGTPjZfbXC6KzYjEcy21R8JEgUnkd1C activePeers:2 Peer Connectedmodule=SYNC
2021-04-16|06:58:37.232 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH direction:Outbound multiAddr:/ip4/103.29.70.78/tcp/28130/p2p/16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH activePeers:3 Peer Connectedmodule=SYNC
2021-04-16|06:58:37.420 [INFO |/p2p/synch/handshake.go:83]           16Uiu2HAmRtp5CjNv3WvPYuh7kNXXZQDYegwFFeDH9vWY3JY4JS1W direction:Outbound multiAddr:/ip4/149.248.8.175/tcp/28230/p2p/16Uiu2HAmRtp5CjNv3WvPYuh7kNXXZQDYegwFFeDH9vWY3JY4JS1W  (Relay)module=SYNC
2021-04-16|06:58:38.752 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAmKNJUYyuH9AkSeY8ytRNy79JRrDeNtupYRx5ZJcs9gZeE direction:Outbound multiAddr:/ip4/8.210.117.240/tcp/28130/p2p/16Uiu2HAmKNJUYyuH9AkSeY8ytRNy79JRrDeNtupYRx5ZJcs9gZeE activePeers:5 Peer Connectedmodule=SYNC
2021-04-16|06:58:40.220 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 direction:Outbound multiAddr:/ip4/47.244.17.119/tcp/28130/p2p/16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 activePeers:6 Peer Connectedmodule=SYNC
2021-04-16|06:58:40.633 [INFO |/services/common/progresslog/blocklogger.go:111] Processed 1 block in the last 12.45s (1 transaction, order uncertainty, 2021-04-16 02:23:40 +0800 CST)module=blkmanager
2021-04-16|06:58:44.776 [ERROR|/p2p/synch/inventory.go:25]                      Failed to reset stream with protocol /qitmeer/req/inventory/1/ssz_snappy,unexpected flagmodule=SYNC
2021-04-16|06:58:46.105 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAm3gKfGGkHCikustscQ35CMPqTwg3iHWT7o4HzMF5oBXSh direction:Outbound multiAddr:/ip4/43.239.159.191/tcp/28130/p2p/16Uiu2HAm3gKfGGkHCikustscQ35CMPqTwg3iHWT7o4HzMF5oBXSh activePeers:7 Peer Connectedmodule=SYNC
2021-04-16|06:58:50.292 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAm1ohBtMhkuEVPCc3ZJYVXcUT87cFQJupG9N5VuBPHKgLP direction:Outbound multiAddr:/ip4/148.70.34.136/tcp/28130/p2p/16Uiu2HAm1ohBtMhkuEVPCc3ZJYVXcUT87cFQJupG9N5VuBPHKgLP activePeers:8 Peer Connectedmodule=SYNC
2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAm44jzg4Xzz7uUujVyj9HsqqBmp3NmREHU2FcJTjngw3id direction:Outbound multiAddr:/ip4/104.224.177.38/tcp/28130/p2p/16Uiu2HAm44jzg4Xzz7uUujVyj9HsqqBmp3NmREHU2FcJTjngw3id activePeers:9 Peer Connectedmodule=SYNC
2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAmKxB76C3dhkyRw18pZiRQo8Tm5mVwrVidLQtvsNHoUUN8 Peer Disconnected,activePeers:8module=SYNC
2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 Peer Disconnected,activePeers:7module=SYNC
2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAkvMptnSRza7PoZGGTPjZfbXC6KzYjEcy21R8JEgUnkd1C Peer Disconnected,activePeers:6module=SYNC
2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH Peer Disconnected,activePeers:5module=SYNC
2021-04-16|06:58:52.755 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAmKNJUYyuH9AkSeY8ytRNy79JRrDeNtupYRx5ZJcs9gZeE Peer Disconnected,activePeers:4module=SYNC
2021-04-16|06:58:55.384 [WARN |/p2p/synch/graphstate.go:85]                     context deadline exceeded          module=SYNC
2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:stream resetmodule=SYNC
2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:stream resetmodule=SYNC
2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):stream resetmodule=SYNC
2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:i/o deadline reachedmodule=SYNC
2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):stream resetmodule=SYNC
2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):i/o deadline reachedmodule=SYNC
2021-04-16|06:58:55.933 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:stream resetmodule=SYNC
2021-04-16|06:58:55.933 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:stream resetmodule=SYNC
2021-04-16|06:58:55.933 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):stream resetmodule=SYNC
2021-04-16|06:58:55.933 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):stream resetmodule=SYNC
2021-04-16|06:59:00.549 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 direction:Outbound multiAddr:/ip4/47.244.17.119/tcp/28130/p2p/16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 activePeers:5 Peer Connectedmodule=SYNC
2021-04-16|06:59:01.184 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH direction:Outbound multiAddr:/ip4/103.29.70.78/tcp/28130/p2p/16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH activePeers:6 Peer Connectedmodule=SYNC

```

**Comment by lochjin** [comment]
> After upgraded to the latest version `5ddabc2` and restarted the crashed qitmeer, the consensus lost.
> 
> ### The consensus lost
> _2021-04-16|06:58:40.633 [INFO |/services/common/progresslog/blocklogger.go:111] Processed 1 block in the last 12.45s (1 transaction, **order uncertainty**, 2021-04-16 02:23:40 +0800 CST)module=blkmanager_
> 
> The block DAG into a error status at `[594,96351,96413]`
> 
> ```
> 594,96351,96413,16Uiu2HAmVMUdpBogwrZKc36RwMkiHdNBYvczNif6sT8398rk3kfY,100000,0.10.0+dev-5ddabc2,30,local
> ```
> 
> ### Log
> ```
> 2021-04-16|06:58:28.000 [INFO |qitmeerd/qitmeerd.go:62]      System info                        Qitmeer Version=0.10.0+dev-5ddabc2 Go version=go1.16
> 2021-04-16|06:58:28.000 [INFO |qitmeerd/qitmeerd.go:63]      System info                        Home dir=/Users/alex/data/qitmeer/0.10-local
> 2021-04-16|06:58:28.000 [INFO |/services/common/db.go:29]    Loading block database             dbPath=/Users/alex/data/qitmeer/0.10-local/data/mixnet/blocks_ffldb
> 2021-04-16|06:58:28.035 [INFO |/database/ffldb/reconcile.go:91] Detected unclean shutdown - Repairing...module=database
> 2021-04-16|06:58:28.039 [INFO |/database/ffldb/reconcile.go:96] Database sync complete             module=database
> 2021-04-16|06:58:28.039 [INFO |/services/common/db.go:49]       Block database loaded
> 2021-04-16|06:58:28.127 [INFO |/p2p/options.go:110]             Start Peers from:/Users/alex/data/qitmeer/0.10-local/data/mixnet/peerstoremodule=P2P
> 2021-04-16|06:58:28.170 [INFO |/node/node.go:131]               Starting Server                    module=node
> 2021-04-16|06:58:28.172 [INFO |/node/full.go:112]               Transaction index is enabled       module=node
> 2021-04-16|06:58:28.177 [INFO |/node/full.go:116]               Address index is enabled           module=node
> 2021-04-16|06:58:28.177 [INFO |/core/blockdag/phantom.go:43]    anticone size:6                    module=blockdag
> 2021-04-16|06:58:28.189 [INFO |/core/blockchain/blockchain.go:496] Loading dag ...                    module=blockchain
> 2021-04-16|06:58:29.772 [INFO |/core/blockchain/blockchain.go:510] Dag loaded:loadTime=1.582815041s   module=blockchain
> 2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:362] DAG Type:phantom                   module=blockchain
> 2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:363] Blockchain database version        module=blockchain chain=7 compression=1 index=1
> 2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:367] Chain state:totaltx=98436 tipsNum=5 mainOrder=98410 total=98418module=blockchain
> 2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:371] hash=665b9e86511c01fc1ecf3fba77c384c0a3d4c355b316f2c0e43de9104d73c5bd,order=uncertainty,work=13021961124926484249272015module=blockchain
> 2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:371] hash=c21f2f114e84ed5ecc05325770abcbe9d0f517342e171e2e4d0f939524e8c5a3,order=uncertainty,work=13040954545946961780708651module=blockchain
> 2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:371] hash=00000265f4cffd307482749124ac7f905fe5a0b785d8437032979284fc208504,order=98410,work=13046688408896539903688257module=blockchain
> 2021-04-16|06:58:32.535 [INFO |/core/blockchain/blockchain.go:371] hash=00888629ae9379c263eeb1fff343484cdc3efdf2a3d6acc87bc4dd6fe62b63d3,order=uncertainty,work=12813391860135580225320356module=blockchain
> 2021-04-16|06:58:32.536 [INFO |/core/blockchain/blockchain.go:371] hash=a1a18b16cbc72e980ebd9e9910178f483ac826757134c6bf9115deae5d65edce,order=uncertainty,work=12926277286955399406386273module=blockchain
> 2021-04-16|06:58:32.536 [INFO |/services/zmq/zmqdefaultnotify.go:17] ZMQ:Not Supported                  module=zmq
> 2021-04-16|06:58:32.536 [INFO |/services/tx/txmanager.go:40]         Starting tx manager                module=txmanager
> 2021-04-16|06:58:32.536 [INFO |/p2p/service.go:93]                   P2P Service Start                  module=P2P
> 2021-04-16|06:58:32.536 [INFO |/p2p/synch/peersync.go:50]            P2P PeerSync Start                 module=SYNC
> 2021-04-16|06:58:32.580 [INFO |/p2p/service.go:701]                  Node started p2p server:multiAddr=/ip4/10.0.0.6/tcp/28130/p2p/16Uiu2HAmVMUdpBogwrZKc36RwMkiHdNBYvczNif6sT8398rk3kfYmodule=P2P
> 2021-04-16|06:58:32.580 [INFO |/p2p/rebroadcast.go:38]               Starting Rebroadcast               module=P2P
> 2021-04-16|06:58:32.593 [INFO |/rpc/rpcserver.go:247]                RPC server listening on addr:0.0.0.0:10101module=RPC
> 2021-04-16|06:58:35.831 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAmKxB76C3dhkyRw18pZiRQo8Tm5mVwrVidLQtvsNHoUUN8 direction:Outbound multiAddr:/ip4/185.186.147.152/tcp/28130/p2p/16Uiu2HAmKxB76C3dhkyRw18pZiRQo8Tm5mVwrVidLQtvsNHoUUN8 activePeers:1 Peer Connectedmodule=SYNC
> 2021-04-16|06:58:35.831 [INFO |/p2p/synch/peersync.go:262]           Syncing to state (99006,96351,96413,99022,2) from peer 16Uiu2HAmKxB76C3dhkyRw18pZiRQo8Tm5mVwrVidLQtvsNHoUUN8 cur graph state:(98410,95813,95874,98418,1)module=SYNC
> 2021-04-16|06:58:36.374 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAkvMptnSRza7PoZGGTPjZfbXC6KzYjEcy21R8JEgUnkd1C direction:Outbound multiAddr:/ip4/149.248.8.175/tcp/28130/p2p/16Uiu2HAkvMptnSRza7PoZGGTPjZfbXC6KzYjEcy21R8JEgUnkd1C activePeers:2 Peer Connectedmodule=SYNC
> 2021-04-16|06:58:37.232 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH direction:Outbound multiAddr:/ip4/103.29.70.78/tcp/28130/p2p/16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH activePeers:3 Peer Connectedmodule=SYNC
> 2021-04-16|06:58:37.420 [INFO |/p2p/synch/handshake.go:83]           16Uiu2HAmRtp5CjNv3WvPYuh7kNXXZQDYegwFFeDH9vWY3JY4JS1W direction:Outbound multiAddr:/ip4/149.248.8.175/tcp/28230/p2p/16Uiu2HAmRtp5CjNv3WvPYuh7kNXXZQDYegwFFeDH9vWY3JY4JS1W  (Relay)module=SYNC
> 2021-04-16|06:58:38.752 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAmKNJUYyuH9AkSeY8ytRNy79JRrDeNtupYRx5ZJcs9gZeE direction:Outbound multiAddr:/ip4/8.210.117.240/tcp/28130/p2p/16Uiu2HAmKNJUYyuH9AkSeY8ytRNy79JRrDeNtupYRx5ZJcs9gZeE activePeers:5 Peer Connectedmodule=SYNC
> 2021-04-16|06:58:40.220 [INFO |/p2p/synch/handshake.go:87]           16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 direction:Outbound multiAddr:/ip4/47.244.17.119/tcp/28130/p2p/16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 activePeers:6 Peer Connectedmodule=SYNC
> 2021-04-16|06:58:40.633 [INFO |/services/common/progresslog/blocklogger.go:111] Processed 1 block in the last 12.45s (1 transaction, order uncertainty, 2021-04-16 02:23:40 +0800 CST)module=blkmanager
> 2021-04-16|06:58:44.776 [ERROR|/p2p/synch/inventory.go:25]                      Failed to reset stream with protocol /qitmeer/req/inventory/1/ssz_snappy,unexpected flagmodule=SYNC
> 2021-04-16|06:58:46.105 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAm3gKfGGkHCikustscQ35CMPqTwg3iHWT7o4HzMF5oBXSh direction:Outbound multiAddr:/ip4/43.239.159.191/tcp/28130/p2p/16Uiu2HAm3gKfGGkHCikustscQ35CMPqTwg3iHWT7o4HzMF5oBXSh activePeers:7 Peer Connectedmodule=SYNC
> 2021-04-16|06:58:50.292 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAm1ohBtMhkuEVPCc3ZJYVXcUT87cFQJupG9N5VuBPHKgLP direction:Outbound multiAddr:/ip4/148.70.34.136/tcp/28130/p2p/16Uiu2HAm1ohBtMhkuEVPCc3ZJYVXcUT87cFQJupG9N5VuBPHKgLP activePeers:8 Peer Connectedmodule=SYNC
> 2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAm44jzg4Xzz7uUujVyj9HsqqBmp3NmREHU2FcJTjngw3id direction:Outbound multiAddr:/ip4/104.224.177.38/tcp/28130/p2p/16Uiu2HAm44jzg4Xzz7uUujVyj9HsqqBmp3NmREHU2FcJTjngw3id activePeers:9 Peer Connectedmodule=SYNC
> 2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAmKxB76C3dhkyRw18pZiRQo8Tm5mVwrVidLQtvsNHoUUN8 Peer Disconnected,activePeers:8module=SYNC
> 2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 Peer Disconnected,activePeers:7module=SYNC
> 2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAkvMptnSRza7PoZGGTPjZfbXC6KzYjEcy21R8JEgUnkd1C Peer Disconnected,activePeers:6module=SYNC
> 2021-04-16|06:58:51.849 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH Peer Disconnected,activePeers:5module=SYNC
> 2021-04-16|06:58:52.755 [INFO |/p2p/synch/handshake.go:156]                     peer:16Uiu2HAmKNJUYyuH9AkSeY8ytRNy79JRrDeNtupYRx5ZJcs9gZeE Peer Disconnected,activePeers:4module=SYNC
> 2021-04-16|06:58:55.384 [WARN |/p2p/synch/graphstate.go:85]                     context deadline exceeded          module=SYNC
> 2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:stream resetmodule=SYNC
> 2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:stream resetmodule=SYNC
> 2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):stream resetmodule=SYNC
> 2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:i/o deadline reachedmodule=SYNC
> 2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):stream resetmodule=SYNC
> 2021-04-16|06:58:55.385 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):i/o deadline reachedmodule=SYNC
> 2021-04-16|06:58:55.933 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:stream resetmodule=SYNC
> 2021-04-16|06:58:55.933 [WARN |/p2p/synch/sync.go:256]                          Failed to handle p2p RPC:stream resetmodule=SYNC
> 2021-04-16|06:58:55.933 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):stream resetmodule=SYNC
> 2021-04-16|06:58:55.933 [WARN |/p2p/synch/sync.go:279]                          Process error (ErrStreamWrite):stream resetmodule=SYNC
> 2021-04-16|06:59:00.549 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 direction:Outbound multiAddr:/ip4/47.244.17.119/tcp/28130/p2p/16Uiu2HAkvbqwZ4HYrs6DZVwoCuWaNNLtdDSUmzKQwWxPgnrzsjz2 activePeers:5 Peer Connectedmodule=SYNC
> 2021-04-16|06:59:01.184 [INFO |/p2p/synch/handshake.go:87]                      16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH direction:Outbound multiAddr:/ip4/103.29.70.78/tcp/28130/p2p/16Uiu2HAkxH285b3xo2wxp3A1rodskqj8kwJUWyTmYdbLjT8uP8FH activePeers:6 Peer Connectedmodule=SYNC
> ```

You can cleanup your block data base by '--cleanup', because of the new version number has been upgraded.

**Comment by lochjin** [comment]
> ### Qitmeer version
> qitmeer version 0.10.0+dev-cc570c5 (Go version go1.16)
> 
> although the issued version `cc570c` is not the latest version `5ddabc`, but the nil pointer issue should exist before `f2305f`, see #474 also.
> 
> * [5ddabc2](https://github.com/Qitmeer/qitmeer/commit/5ddabc2b32301d66c570add5a6b8b6e09542cb68)
> * [cc570c5](https://github.com/Qitmeer/qitmeer/commit/cc570c526062facf2cdc7c90207ce89c24539fea)
> * [f2305f2](https://github.com/Qitmeer/qitmeer/commit/f2305f24549b70b643e11330493d725ea96ba52c)
> 
> ### Error log
> ```
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1b4121c]
> 
> goroutine 3314967 [running]:
> github.com/Qitmeer/qitmeer/p2p/peers.(*Peer).GraphState(0x0, 0x0)
>         /Users/alex/work/qitmeer/qitmeer/p2p/peers/peer.go:373 +0x3c
> github.com/Qitmeer/qitmeer/p2p/synch.(*PeerSync).IsCompleteForSyncPeer(0xc00025d000, 0xc0009ae001)
>         /Users/alex/work/qitmeer/qitmeer/p2p/synch/peersync.go:356 +0x57
> github.com/Qitmeer/qitmeer/p2p/synch.(*PeerSync).IsCurrent(0xc00025d000, 0x8737220222786)
>         /Users/alex/work/qitmeer/qitmeer/p2p/synch/peersync.go:344 +0x5a
> github.com/Qitmeer/qitmeer/p2p/synch.(*Sync).handleInventory(0xc000288870, 0xc000648dc0, 0xc000f00000, 0x21337b8, 0xc000648e40)
>         /Users/alex/work/qitmeer/qitmeer/p2p/synch/inventory.go:73 +0x5b
> github.com/Qitmeer/qitmeer/p2p/synch.(*Sync).inventoryHandler(0xc000288870, 0x2133828, 0xc0028fa420, 0x1e260e0, 0xc000648dc0, 0x21406e8, 0xc00f7875c0, 0x0)
>         /Users/alex/work/qitmeer/qitmeer/p2p/synch/inventory.go:58 +0x11d
> github.com/Qitmeer/qitmeer/p2p/synch.RegisterRPC.func1(0x21406e8, 0xc00f7875c0)
>         /Users/alex/work/qitmeer/qitmeer/p2p/synch/sync.go:255 +0x74a
> github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).SetStreamHandler.func1(0xc004fcf890, 0x23, 0x47b7360, 0xc00f7875c0, 0x1, 0x0)
>         /Users/alex/go/pkg/mod/github.com/libp2p/go-libp2p@v0.11.0/p2p/host/basic/basic_host.go:566 +0xa2
> created by github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).newStreamHandler
>         /Users/alex/go/pkg/mod/github.com/libp2p/go-libp2p@v0.11.0/p2p/host/basic/basic_host.go:416 +0x665
> ```

Does your machine have enough memory ?

**Comment by dindinw** [comment]
fixed by #478
